src/modules/wasi/x86-64-linux/WspOneModule-x86-64-linux.v3

```diff
@@ -169,11 +315,19 @@ class X86_64Linux_WspOneModule extends WspOneModule {
 
 		var flag_r = (base_rights & (1 << WspOneRights.fd_read.tag)) != 0;
 		var flag_w = (base_rights & (1 << WspOneRights.fd_write.tag)) != 0;
+                flag_w |= (base_rights & (1 << WspOneRights.fd_filestat_set_size.tag)) != 0;
+                flag_w |= (base_rights & (1 << WspOneRights.fd_allocate.tag)) != 0;
 		var sysflags = if(flag_w, if(flag_r, LinuxConst.O_RDWR, LinuxConst.O_WRONLY), LinuxConst.O_RDONLY);
 
-		if ((oflags & (1u << WspOneFdFlags.creat.tag)) != 0) sysflags |= LinuxConst.O_CREAT;
-		if ((oflags & (1u << WspOneFdFlags.trunc.tag)) != 0) sysflags |= LinuxConst.O_TRUNC;
-		if ((oflags & (1u << WspOneFdFlags.excl.tag)) != 0) sysflags |= LinuxConst.O_EXCL;
+		if ((oflags & (1u << WspOneOFlags.creat.tag)) != 0) sysflags |= LinuxConst.O_CREAT;
+		if ((oflags & (1u << WspOneOFlags.directory.tag)) != 0) sysflags |= LinuxConst.O_DIRECTORY;
+		if ((oflags & (1u << WspOneOFlags.excl.tag)) != 0) sysflags |= LinuxConst.O_EXCL;
+		if ((oflags & (1u << WspOneOFlags.trunc.tag)) != 0) sysflags |= LinuxConst.O_TRUNC;
+		if ((fflags & (1u << WspOneFdFlags.append.tag)) != 0) sysflags |= LinuxConst.O_APPEND;
+//		if ((fflags & (1u << WspOneFdFlags.dsync.tag)) != 0) sysflags |= LinuxConst.O_DSYNC;
```

Comment:
Are these TODOs are unused? If they are unused, we should drop them.

Reply:
These are more like TODOs.